### PR TITLE
Various libnetwork cleanups

### PR DESCRIPTION
**Pull request by cpuguy83**
See commits for details.
Related to #42471

### CODE REVIEW

**Comment by thaJeztah** [comment]
❤️❤️ thanks! will have a closer look tomorrow, but SGTM

**Comment by thaJeztah** [comment]
Looks like CI is failing; probably some changes needed for the move of docker-proxy to a different location;

```
=== RUN   TestNetworkNat
    nat_test.go:30: assertion failed: error is not nil: Error response from daemon: driver failed programming external connectivity on endpoint server-TestNetworkNat (37609f18f3c616cad5f6e55b316778d0d083ab2c341d4c7900efc35b8f9b171d): exec: "docker-proxy": executable file not found in $PATH
```

File: hack/make/dynbinary-proxy [hunk 1]
```diff
@@ -9,7 +9,7 @@ set -e
 	export BUILDFLAGS=("${BUILDFLAGS[@]/osusergo /}")     # ditto for osusergo
 	export BUILDFLAGS=("${BUILDFLAGS[@]/static_build /}") # we're not building a "static" binary here
 
-	GO_PACKAGE='github.com/docker/docker/libnetwork/cmd/proxy'
+	GO_PACKAGE='github.com/docker/docker/cmd/dockeer-proxy'
```
Comment (thaJeztah) [inline hack/make/dynbinary-proxy:12]: Ah! Guess this is the cause;

```suggestion
	GO_PACKAGE='github.com/docker/docker/cmd/docker-proxy'
```

**Comment by thaJeztah** [comment]
I rebased, fixed the typo (to see if that fixes the issue), and removed the `proxy.installer` script (should no longer be needed)

**Review by tianon — APPROVED** [review]
:eyes:

**Review by samuelkarp — APPROVED** [review]
LGTM

**Comment by samuelkarp** [comment]
More flakes of https://github.com/moby/moby/issues/42459 in CI, rerunning.

**Review by thaJeztah — APPROVED** [review]
LGTM

**Pull request merged**